### PR TITLE
Logging structure - add extraInfo: {}

### PR DESCRIPTION
**Pull request by antobinary**
Added `extraInfo` for additional data, turned the logs from client to server and from client to nginx (external) into json format
Sample:

I20190702-16:52:21.295(0)? info: {"logCode":"sip_js_session_progress","logDescription":"Audio call session progress update","extraInfo":{"validUser":"valid"},"userInfo":{"sessionToken":"yhyqjc9wqqd29uqw","meetingId":"183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1562082413090","requesterUserId":"w_n7tvzo152qe3","fullname":"_________","confname":"Demo Meeting","externUserID":"w_n7tvzo152qe3","uniqueClientSession":"r67zew"}}

Sample:
Screenshot from 2019-07-02 12-55-03

### CODE REVIEW

File: bigbluebutton-html5/imports/api/audio/client/bridge/sip.js [hunk 1]
```diff
@@ -188,15 +191,15 @@ class SIPSession {
       // translation
       const isSafari = browser().name === 'safari';
 
-      logger.debug('Creating the user agent');
+      logger.debug({ logCode: 'creating_user_agent' }, 'Creating the user agent');
```
Comment (capilkey) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sip.js]: The code should be prepended with "sip_js"

File: bigbluebutton-html5/imports/api/audio/client/bridge/sip.js [hunk 2]
```diff
@@ -188,15 +191,15 @@ class SIPSession {
...
 
       if (this.userAgent && this.userAgent.isConnected()) {
         if (this.userAgent.configuration.hostPortParams === this.hostname) {
-          logger.debug('Reusing the user agent');
+          logger.debug({ logCode: 'reusing_user_agent' }, 'Reusing the user agent');
```
Comment (capilkey) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sip.js]: The code should be prepended with "sip_js"

File: bigbluebutton-html5/imports/api/audio/client/bridge/sip.js [hunk 3]
```diff
@@ -188,15 +191,15 @@ class SIPSession {
...
           resolve(this.userAgent);
           return;
         }
-        logger.debug('different host name. need to kill');
+        logger.debug({ logCode: 'different_host_name' }, 'Different host name. need to kill');
```
Comment (capilkey) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sip.js]: The code should be prepended with "sip_js"

File: bigbluebutton-html5/imports/ui/components/video-provider/component.jsx [hunk 4]
```diff
@@ -990,20 +1192,37 @@ class VideoProvider extends Component {
...
   unshareWebcam() {
     const { userId } = this.props;
-    this.logger('info', 'Unsharing webcam', 'video_provider_unshare_webcam');
+    logger.info({
+      logCode: 'video_provider_unshare_webcam',
+      extraInfo: {
+        topic: 'video',
+      },
+    }, 'Unsharing webcam');
+    this.sharedWebcam = true;
```
Comment (capilkey) [inline bigbluebutton-html5/imports/ui/components/video-provider/component.jsx]: Was this supposed to be added?

File: bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js [hunk 5]
```diff
@@ -126,7 +133,9 @@ export default class KurentoAudioBridge extends BaseAudioBridge {
         await audioContext.setSinkId(value);
         this.media.outputDeviceId = value;
       } catch (err) {
-        logger.error({ logCode: 'audio_kurento_changeoutputdevice_error' }, err);
+        logger.error({ logCode: 'audio_kurento_changeoutputdevice_error', err });
+        logFunc('error', 'SFU audio bridge failed to fetch STUN/TURN info, using default',
+          { logCode: 'audio_kurento_changeoutputdevice_error', err });
```
Comment (capilkey) [inline bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: Why is `logger.error` and `logFunc` called together? Don't they both go to the same place?

File: bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js [hunk 6]
```diff
@@ -11,7 +11,14 @@ const GLOBAL_AUDIO_PREFIX = 'GLOBAL_AUDIO_';
...
 
-  logger[type](`[${topic}] ${JSON.stringify(message)} options:${JSON.stringify(options)}`);
+  logger[type]({
+    logCode: 'TODO_kurento_audio',
+    extraInfo: {
+      topic,
+      message,
+      options,
+    },
+  }, options && options.message && options.message.id ? options.message.id : 'kurento audio');
```
Comment (capilkey) [inline bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: Shouldn't the line be `), message);`? Is the message contained in the options object?

File: bigbluebutton-html5/imports/ui/components/video-provider/component.jsx (hunk 4 above)
Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/video-provider/component.jsx]: Nice catch! :sweat_smile: copy-paste leftover

File: bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js (hunk 6 above)
Comment (antobinary) [reply bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: I was looking for a string to display. A group of messages have a decently descriptive `options.message.id` but not all. For the rest - I add `'kurento audio'` so we have something

Comment (capilkey) [reply bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: Don't they all pass in the `message` parameter though? Seems easiest to just use that

Comment (antobinary) [reply bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: `"message":"Sending message:",`
vs
`options.message.id: "start",` with topic: 'audio'
Neither is great but I find the options.message.id a bit more meaningful

Comment (capilkey) [reply bigbluebutton-html5/imports/api/audio/client/bridge/kurento.js]: I didn't realize that both options were pretty bad.

**Pull request merged**